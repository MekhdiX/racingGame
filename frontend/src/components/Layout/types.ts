export type OwnProps = {
    title?: string,
    transparent?: boolean,
    className?: string,
    verticalAlign?: boolean,
}
