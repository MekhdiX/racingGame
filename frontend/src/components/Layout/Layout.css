.layout {
  min-height: 100vh;
  display: flex;

  &.light {
    background-color: #ffffff;
  }

  &.dark {
    background-color: #000000;
  }

  &__app-name {
    font-size: 3em;
    padding: 4px 8px;
    border-radius: 8px;
    display: inline-block;
    line-height: 1;
    font-weight: 700;
    margin-bottom: 12px;
    text-transform: uppercase;
    position: relative;

    &.light {
      background-color: #000000;
      color: #ffffff;
    }

    &.dark,
    &.dark:hover {
      background-color: #fff;
      color: #000000;
      box-shadow: 0 0 6px #ffffffcf;
    }

    &:hover {
      color: #ffffff;
      opacity: 0.8;
    }
  }

  &__app-name-wrapper {
    margin-top: 50px;
    text-align: center;
  }

  &__app-name[data-highlightword]:before {
    content: attr(data-highlightword);
    color: red;
    position: absolute;
    text-shadow: 0 0 5px #000000;
  }

  .ui.layout__container {
    &_valign {
      display: flex;
      flex-direction: column;
      .layout__main {
        flex-grow: 1;
        display: flex;
        align-items: center;
      }
    }
  }
}

.ui.menu.layout__main-menu {
  margin-bottom: 30px;
}

h1.ui.header.layout__header {
  font-size: 2.7em;

  .dark & {
    color: #ffffff;
  }
}

.main {
  background-color: #ececec;
  border-radius: 4px;
  box-shadow: 0 6px 8px rgba(0, 0, 0, 0.1);
  padding: 15px;

  &_transparent {
    background-color: transparent;
    box-shadow: none;
  }

  .ui.green.button {
    .light & {
      color: #ffffff;
    }
    .dark & {
      color: #000000;
    }
  }
  .ui.green.button:hover {
    color: #ffffff;
  }
}
