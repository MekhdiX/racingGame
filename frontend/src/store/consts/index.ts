// Auth actions
export const SET_AUTH = 'SET_AUTH';
export const SIGN_IN = 'SIGN_IN';
export const SIGN_IN_YANDEX_OAUTH = 'SIGN_IN_YANDEX_OAUTH';
export const SIGN_UP = 'SIGN_UP';
export const LOG_OUT = 'LOG_OUT';

//User actions
export const SET_USER_INFO = 'SET_USER_INFO';
export const GET_USER = 'GET_USER';
export const CHANGE_USER_INFO = 'CHANGE_USER_INFO';
export const CHANGE_AVATAR = 'CHANGE_AVATAR';

// SSR actions
export const SET_SSR = 'SET_SSR';

// Theme actions
export const GET_THEME = 'GET_THEME';
export const SET_THEME = 'SET_THEME';

// Leaderboard actions
export const SET_LEADERBOARD = 'SET_LEADERBOARD';

// Feedback actions
export const SEND_FEEDBACK = 'SEND_FEEDBACK';

export const SET_FORUM_TOPIC = 'FETCH_FORUM_TOPIC';