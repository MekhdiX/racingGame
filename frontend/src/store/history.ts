import { createBrowserHistory, createMemoryHistory } from 'history';
import isServer from 'utils/isServerEnvCheker';

export default !isServer ? createBrowserHistory() : createMemoryHistory();
