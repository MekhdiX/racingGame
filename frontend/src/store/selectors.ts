import { AppState } from './types';

export const getAuthSelector = (state: AppState) => state.auth;

export const getUserSelector = (state: AppState) => state.user;

export const getSSRSelector = (state: AppState) => state.ssr;

export const getThemeSelector = (state: AppState) => state.theme;

export const getLeaderboardSelector = (state: AppState) => state.leaderboard;

export const getForumTopicsSelector = (state: AppState) => state.forumTopic;
