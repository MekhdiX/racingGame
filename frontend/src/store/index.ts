import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { routerMiddleware } from 'connected-react-router';
import history from './history';
import reducer from './reducers';
import { composeWithDevTools } from 'redux-devtools-extension';

const enhancer = applyMiddleware(thunk, routerMiddleware(history));

const store = createStore(reducer, composeWithDevTools(enhancer));

export default store;
