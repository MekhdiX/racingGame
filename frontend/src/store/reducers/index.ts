import { combineReducers } from 'redux';
import { connectRouter } from 'connected-react-router';

import history from '../history';
import auth from './auth';
import user from './user';
import ssr from './ssr';
import theme from './theme';
import leaderboard from './leaderboard';
import forumTopic from './forumTopic';

const reducer = combineReducers({
  router: connectRouter(history),
  auth,
  user,
  ssr,
  theme,
  leaderboard,
  forumTopic
});

export default reducer;
