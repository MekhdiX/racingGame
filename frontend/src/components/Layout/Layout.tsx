import './Layout.css';

import { ROUTES } from 'common/consts';
import { t } from 'common/dictionary';
import MainMenu from 'components/MainMenu';
import React, { FC, PropsWithChildren } from 'react';
import { Link } from 'react-router-dom';
import { Container, Header } from 'semantic-ui-react';

import { OwnProps } from './types';
import { useSelector } from 'react-redux';
import { getThemeSelector } from 'store/selectors';

const Layout: FC<OwnProps> = (props: PropsWithChildren<OwnProps>) => {
  const { className, verticalAlign, title, transparent, children } = props;

  const theme = useSelector(getThemeSelector);

  return (
    <div className={`layout ${className ?? ''} ${theme}`}>
      <Container className={verticalAlign ? 'layout__container_valign' : ''}>
        <div className="layout__app-name-wrapper">
          <Link to={ROUTES.ROOT} className={`layout__app-name ${theme}`} data-highlightword={t('appName').split(' ')[0]}>
            {t('appName')}
          </Link>
        </div>
        <MainMenu />
        {title && (
          <Header as="h1" className="layout__header">
            {title}
          </Header>
        )}
        <div className={`layout__main main ${transparent ? 'main_transparent' : ''}`}>{children}</div>
      </Container>
    </div>
  );
};
export default Layout;
