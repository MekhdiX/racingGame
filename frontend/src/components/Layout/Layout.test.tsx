import Layout from 'components/Layout/Layout';
import { ConnectedRouter } from 'connected-react-router';
import * as React from 'react';
import { Provider } from 'react-redux';
import * as renderer from 'react-test-renderer';
import store from 'store';
import history from 'store/history';

describe(`Layout Component`, () => {
  it(`renders correctly`, () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ConnectedRouter history={history}>
            <Layout>
              <div>Hello world</div>
            </Layout>
          </ConnectedRouter>
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`renders correctly with title prop`, () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ConnectedRouter history={history}>
            <Layout title="Title">
              <div>Hello world</div>
            </Layout>
          </ConnectedRouter>
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`renders correctly with all props`, () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ConnectedRouter history={history}>
            <Layout title="Title" transparent className="class-name" verticalAlign>
              <div>Hello world</div>
            </Layout>
          </ConnectedRouter>
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
