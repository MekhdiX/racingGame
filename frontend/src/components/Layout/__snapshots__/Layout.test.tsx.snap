// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component renders correctly 1`] = `
<div
  className="layout  light"
>
  <div
    className="ui container"
  >
    <div
      className="layout__app-name-wrapper"
    >
      <a
        className="layout__app-name light"
        data-highlightword="Casablanca"
        href="/"
        onClick={[Function]}
      >
        Casablanca racing
      </a>
    </div>
    <div
      className="ui main-menu light layout__main-menu menu"
    >
      <a
        className="item main-menu__item"
        onClick={[Function]}
      >
        Играть
      </a>
      <div
        className="right menu"
      >
        <div
          className="toggle_theme"
        >
          <i
            aria-hidden="true"
            className="yellow sun large icon toggle_theme__icon"
            onClick={[Function]}
          />
          <div
            className="ui fitted toggle checkbox"
            color="black"
            onChange={[Function]}
            onClick={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
          >
            <input
              checked={false}
              className="hidden"
              readOnly={true}
              tabIndex={0}
              type="checkbox"
            />
            <label />
          </div>
          <i
            aria-hidden="true"
            className="moon large icon toggle_theme__icon"
            onClick={[Function]}
          />
        </div>
        <a
          className="item main-menu__item"
          onClick={[Function]}
        >
          Войти
        </a>
        <a
          className="item main-menu__item"
          onClick={[Function]}
        >
          Зарегистрироваться
        </a>
      </div>
    </div>
    <div
      className="layout__main main "
    >
      <div>
        Hello world
      </div>
    </div>
  </div>
</div>
`;

exports[`Layout Component renders correctly with all props 1`] = `
<div
  className="layout class-name light"
>
  <div
    className="ui container layout__container_valign"
  >
    <div
      className="layout__app-name-wrapper"
    >
      <a
        className="layout__app-name light"
        data-highlightword="Casablanca"
        href="/"
        onClick={[Function]}
      >
        Casablanca racing
      </a>
    </div>
    <div
      className="ui main-menu light layout__main-menu menu"
    >
      <a
        className="item main-menu__item"
        onClick={[Function]}
      >
        Играть
      </a>
      <div
        className="right menu"
      >
        <div
          className="toggle_theme"
        >
          <i
            aria-hidden="true"
            className="yellow sun large icon toggle_theme__icon"
            onClick={[Function]}
          />
          <div
            className="ui fitted toggle checkbox"
            color="black"
            onChange={[Function]}
            onClick={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
          >
            <input
              checked={false}
              className="hidden"
              readOnly={true}
              tabIndex={0}
              type="checkbox"
            />
            <label />
          </div>
          <i
            aria-hidden="true"
            className="moon large icon toggle_theme__icon"
            onClick={[Function]}
          />
        </div>
        <a
          className="item main-menu__item"
          onClick={[Function]}
        >
          Войти
        </a>
        <a
          className="item main-menu__item"
          onClick={[Function]}
        >
          Зарегистрироваться
        </a>
      </div>
    </div>
    <h1
      className="ui header layout__header"
    >
      Title
    </h1>
    <div
      className="layout__main main main_transparent"
    >
      <div>
        Hello world
      </div>
    </div>
  </div>
</div>
`;

exports[`Layout Component renders correctly with title prop 1`] = `
<div
  className="layout  light"
>
  <div
    className="ui container"
  >
    <div
      className="layout__app-name-wrapper"
    >
      <a
        className="layout__app-name light"
        data-highlightword="Casablanca"
        href="/"
        onClick={[Function]}
      >
        Casablanca racing
      </a>
    </div>
    <div
      className="ui main-menu light layout__main-menu menu"
    >
      <a
        className="item main-menu__item"
        onClick={[Function]}
      >
        Играть
      </a>
      <div
        className="right menu"
      >
        <div
          className="toggle_theme"
        >
          <i
            aria-hidden="true"
            className="yellow sun large icon toggle_theme__icon"
            onClick={[Function]}
          />
          <div
            className="ui fitted toggle checkbox"
            color="black"
            onChange={[Function]}
            onClick={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
          >
            <input
              checked={false}
              className="hidden"
              readOnly={true}
              tabIndex={0}
              type="checkbox"
            />
            <label />
          </div>
          <i
            aria-hidden="true"
            className="moon large icon toggle_theme__icon"
            onClick={[Function]}
          />
        </div>
        <a
          className="item main-menu__item"
          onClick={[Function]}
        >
          Войти
        </a>
        <a
          className="item main-menu__item"
          onClick={[Function]}
        >
          Зарегистрироваться
        </a>
      </div>
    </div>
    <h1
      className="ui header layout__header"
    >
      Title
    </h1>
    <div
      className="layout__main main "
    >
      <div>
        Hello world
      </div>
    </div>
  </div>
</div>
`;
